lib/ReactViews/FeatureInfo/FeatureInfoSection.jsx

```diff
@@ -306,13 +306,32 @@ function getPropertyValuesForFeature(feature, clock, formats) {
     // If they have bad keys, fix them.
     // If they have formatting, apply it.
     const properties = feature.currentProperties || propertyGetTimeValues(feature.properties, clock);
-    const result = replaceBadKeyCharacters(properties);
+    // Try JSON.parse on values that look like JSON arrays or objects
+    let result = parseValues(properties);
+    result = replaceBadKeyCharacters(result);
     if (defined(formats)) {
         applyFormatsInPlace(result, formats);
     }
     return result;
 }
 
+function parseValues(properties) {
```

Comment:
This is conceptually the wrong place to be modifying feature info properties (it's in the view layer, not the model layer), but I'm not sure if there's anything in the model layer that reads all feature info properties.

Reply:
I think it's ok.  Or maybe it's a symptom of a bigger problem.

Currently, the whole "feature info template" thing is something the UI does, for better or worse (that's the potentially "bigger problem" I'm referring to above).  So we can think of this `parseValues` function as just a step in the process of presenting data to the feature info template.  It doesn't actually modify the properties of the feature; clearly it would be pretty dodgy if the UI were quietly modifying data in the Models layer.

So I don't think it's a major problem.  Is there an aspect to it that I'm missing?

Reply:
That makes sense. There's no problem with leaving it in the view layer.